### PR TITLE
document inverse problem and introduce notations + simplify code

### PR DESCRIPTION
**Pull request by agramfort**
@maikia @hoechenberger @lucyleeow let me know what you think. I tried to simplify
and add some explanations on the inverse problem.

### CODE REVIEW

**Comment by agramfort** [comment]
some suggestion. Given the notations now introduced I would rename the matrix lead_field to L in the .npz files.

File: meg_starting_kit.ipynb [hunk 1]
```diff
@@ -1025,97 +1055,77 @@
...
+    "*Remark:* Note that we are scaling all the 'lead_fields' by 1e8 in the `get_leadfields` function. That is to avoid having too small numbers given to the estimator.\n",
     "\n",
-    "X_test_mapped = X_test.copy()\n",
-    "X_test_mapped['subject_id'] = X_test_mapped['subject'].map(subj_dict)\n",
-    "# scale to avoid tiny numbers\n",
-    "X_test_mapped.iloc[:, :-2] *= 1e12"
+    "The idea of the Lasso is estimate $z \\in \\mathbb{R}^q$ as:\n",
+    "\n",
+    "$\n",
+    "\\hat{z} \\in \\mathop{\\mathrm{arg\\,min}}_z \\frac{1}{2p} \\| x - Lz \\|_2^2 + \\alpha \\|z\\|_1\n",
```
Comment (lucyleeow) [inline meg_starting_kit.ipynb:1095]: is p here the number of sensors?

File: meg_starting_kit.ipynb [hunk 2]
```diff
@@ -1025,97 +1055,77 @@
...
-    "X_train_mapped.iloc[:, :-2] *= 1e12\n",
+    "We will now construct slightly more complicated estimator which will use 'lead_fields'. First we want to load the 'lead_fields' which are used in our data.\n",
+    "\n",
+    "*Remark:* Note that we are scaling all the 'lead_fields' by 1e8 in the `get_leadfields` function. That is to avoid having too small numbers given to the estimator.\n",
     "\n",
-    "X_test_mapped = X_test.copy()\n",
-    "X_test_mapped['subject_id'] = X_test_mapped['subject'].map(subj_dict)\n",
-    "# scale to avoid tiny numbers\n",
-    "X_test_mapped.iloc[:, :-2] *= 1e12"
+    "The idea of the Lasso is estimate $z \\in \\mathbb{R}^q$ as:\n",
```
Comment (lucyleeow) [inline meg_starting_kit.ipynb:1092]: 'is to solve the inverse problem, estimate...'

Maybe talk about the inverse problem here, as it's confusing when you mention it below, as people might think you are talking about the inverse of this equation

**Review by lucyleeow — COMMENTED** [review]
Just some suggestions and questions. The code is much cleaner

**Comment by lucyleeow** [comment]
I've added my changes to the notebook directly @agramfort

File: meg_starting_kit.ipynb (hunk 1 above)
Comment (lucyleeow) [reply meg_starting_kit.ipynb:1097]: Can you clarify this still?

File: meg_starting_kit.ipynb [hunk 3]
```diff
@@ -159,14 +182,30 @@
...
+    "#### Mathematical foundations of MEG source imaging\n",
+    "\n",
+    "If one denote by $x \\in \\mathbb{R}^p$ the MEG data at one time point, where $p$ is the number of sensors, one has that $x$ is obtained from a linear combination of sources. The linearity is a fact derived from the linearity of Maxwell equations. Assuming the sources can originate from a finite set of $q$ locations in the brain, one has that the data $x$ is related to the source amplitudes $z \\in \\mathbb{R}^q$ via a matrix $L$ of size $p \\times q$, known as the 'lead field matrix', the 'gain' or 'forward operator'. It represents the sensitivity of each sensor to signals from each source location. For example, if the value of $L_{i,j}$ is high, it means that a signal from source $z_j$ would result in a high measurement in sensor $x_i$. Conversely, if the value of $L_{i,j}$ is low, it means that a signal from source $z_j$ would result in a low measurement in sensor $x_i$. \n",
```
Comment (lucyleeow) [inline meg_starting_kit.ipynb:194]: @agramfort is this explanation of lead field correct?

**Comment by lucyleeow** [comment]
Failure is due to problem downloading the data.

**Comment by lucyleeow** [comment]
`download_data.py` works for me fine locally, maybe a travis/OFS issue?

**Comment by agramfort** [comment]
we can restart the build or merge as is. I am not worried

**Review by maikia — COMMENTED** [review]
LGTM. Just a single question:

in the meg_starting_kit notebook, section 'origin of electrical signals in the brain', last sentence: 

> This group signal (from the single-cell dipoles that add up) along with a lot of noise can be recorded by MEG or EEG.

why to add MEG? I thought MEG does not record electric signal?

**Comment by agramfort** [comment]
The signal here should refer to the electromagnetic signal in my mind

**Comment by maikia** [comment]
> The signal here should refer to the electromagnetic signal in my mind

The title of the section is `Origin of electrical signals in the brain` so I would either change it to `origin of electromagnetic signal` and remove the title of the next subsection: `Origin of magnetic signals in the brain` or remove the MEG from the above sentence.

**Comment by agramfort** [comment]
The signal in the brain is electric. These are electric current dipoles. They induce an electromagnetic signal around and outside.

**Comment by maikia** [comment]
thanks @agramfort and @lucyleeow

**Pull request merged**